### DISCUSSION TITLE
WASM vs Rust determinism

### DISCUSSION THREAD

**Discussion by nevermemo** [dimforge/rapier#768] [Q&A] [answered]
Hello,
I am looking for a deterministic physics engine to be used in an upcoming project. For this I will need to create two different systems. One of them will run simulations headless (Rust), the other will have a FE and use WASM (JS). I read both versions can use enhanced-determinism and give the same results with the same initial conditions. But if I run the same simulation in both rust and JS with the same initial parameters (using a snapshot, not JS floats), would it still give the same results?

**Comment by ThierryBerger** [comment, marked as answer]
Yes, cross-platform determinism is possible through `enhanced-determinism`, more info at https://www.rapier.rs/docs/user_guides/rust/determinism.
